### PR TITLE
feat(osc): add rate limiter for osc

### PR DESCRIPTION
**Pull request by krihy**
#### What type of PR is this?

type-feature
module-online-schema-change
#### What this PR does / why we need it:
1. we need to add rate limiter when  transfer data from old table to ghost table
2. also we support modify rate limiter before task terminate
3. add `odc.osc.rate-limit.enabled` to control  font to show rate limit, default is disable

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:
this pr has test pass in cloud

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/service/onlineschemachange/model/RateLimiterConfig.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ * @date 2024-04-28
+ * @since 4.2.4
+ */
+@Data
+public class RateLimiterConfig {
+
+    /**
+     * limit max row size per second
+     */
+    private Integer rowLimit;
```
Comment (yizhouxw) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/onlineschemachange/model/RateLimiterConfig.java:31]: maxRowsPerSecond,
maxMegaBytesPerSecond

File: server/odc-service/src/main/java/com/oceanbase/odc/service/onlineschemachange/model/ThrottleConfig.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ * limitations under the License.
+ */
+package com.oceanbase.odc.service.onlineschemachange.model;
+
+/**
+ * @author yaobin
+ * @date 2024-04-28
+ * @since 4.2.4
+ */
+public interface ThrottleConfig {
```
Comment (yizhouxw) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/onlineschemachange/model/ThrottleConfig.java:23]: it seems concept in oms, may move to oms package

File: server/odc-service/src/main/java/com/oceanbase/odc/service/onlineschemachange/model/RateLimiterConfig.java (hunk 1 above)
Comment (krihy) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/onlineschemachange/model/RateLimiterConfig.java:31]: this filed is in order to keep consistency with `DLM` and front-end has adapt to api,  suggest remain unchanged

File: server/odc-service/src/main/java/com/oceanbase/odc/service/onlineschemachange/model/ThrottleConfig.java (hunk 2 above)
Comment (krihy) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/onlineschemachange/model/ThrottleConfig.java:23]: ok

File: server/odc-server/src/main/java/com/oceanbase/odc/server/web/controller/v2/OscController.java [hunk 3]
```diff
@@ -54,4 +56,11 @@ public SuccessResponse<OscSwapTableVO> swapTable(@PathVariable Long scheduleTask
         return Responses.success(oscService.swapTable(scheduleTaskId));
     }
 
+    @ApiOperation(value = "updateRateLimitConfig", notes = "update osc rate limit config")
+    @RequestMapping(value = "/updateRateLimitConfig", method = RequestMethod.POST)
+    public SuccessResponse<Boolean> updateThrottleConfig(
```
[the remark this answers is not in the document]
Comment (krihy) [reply server/odc-server/src/main/java/com/oceanbase/odc/server/web/controller/v2/OscController.java:61]: ok

**Review by yizhouxw — APPROVED** [review]

**Review by yhilmare — APPROVED** [review]
lgtm

**Review by guowl3 — APPROVED** [review]
LGTM

**Review by LuckyPickleZZ — APPROVED** [review]

**Pull request merged**